src/components/activity/ActivityDirectiveChangelog.svelte

```diff
@@ -0,0 +1,365 @@
+      // e.g. an API update that didn't actually change anything, or creating a preset from the current parameter values
+      return { currentValue: '', name: 'No Changes', previousValue: '' };
+    }
+
+    return {
+      currentValue: firstChange.currentValue,
+      name: changedProperties[0] || '',
+      previousValue: firstChange.previousValue,
+    };
+  }
+
+  async function restoreRevision(revisionId: number) {
+    const { id: activityId, plan_id: planId } = activityDirective;
+    const restored = await effects.restoreActivityFromChangelog(activityId, planId, revisionId, user);
+
+    if (restored) {
+      dispatch('closeChangelog');
+    }
+  }
+
+  onMount(async () => {
+    const { id: activityId, plan_id: planId } = activityDirective;
+    activityRevisions = await effects.getActivityDirectiveChangelog(planId, activityId, user);
+
+    // Get default set of effective arguments
+    const effectiveDefaultArguments = await effects.getEffectiveActivityArguments(
+      modelId,
+      activityType?.name || '',
+      {},
+      user,
+    );
+
+    // Get effective arguments for all revisions by coalescing defaults with args supplied in revision
+    effectiveRevisionArguments = activityRevisions.map(revision => {
+      return {
+        ...effectiveDefaultArguments?.arguments,
+        ...revision.arguments,
+      };
+    });
+
+    // Build a summary of changes by comparing two revisions and their effective arguments
+    activityRevisionChangeMap = activityRevisions.map((activityRevision, i) => {
+      const previousRevision = activityRevisions[i + 1];
+
+      // At some point there will be no previous revision to compute a difference from
+      if (!previousRevision) {
+        return { currentValue: '', name: 'Earliest Known Revision', previousValue: '' };
+      }
+
+      return diffRevisions(
+        { ...activityRevision, arguments: { ...effectiveRevisionArguments[i] } },
+        { ...previousRevision, arguments: { ...effectiveRevisionArguments[i + 1] } },
+      );
+    });
+  });
+</script>
+
+<div class="activity-header">
+  <div class="activity-header-title">
+    <div class="activity-header-title-value st-typography-medium">
+      {activityType?.name}
+    </div>
+  </div>
+  <div>
+    <button
+      class="st-button icon activity-header-changelog"
+      on:click|stopPropagation={() => dispatch('closeChangelog')}
+      use:tooltip={{ content: 'Close Activity Changelog', placement: 'top' }}
+    >
+      <HistoryIcon />
+    </button>
+  </div>
+</div>
+
+<div class="activity-revisions">
+  {#if activityRevisionChangeMap.length}
+    {#each activityRevisions as revision, i}
+      <div class="activity-revision">
+        <div class="date st-typography-medium">{formatDate(revision.changed_at)}</div>
+        <div class="change-summary st-typography-body">
+          {activityRevisionChangeMap[i].name}
+        </div>
+        <div class="changed-by st-typography-label">{revision.changed_by}</div>
```

Comment:
Might want to add an `@` prefix to user to match simulation history and plan snapshot @lklyne 
<img width="418" alt="image" src="https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/f15517c7-dc0b-4ce9-a8ae-85e7d9bb6dc4">